Automattic/jetpack
# Social: Update share status to start polling on retry and show spinner

## Proposed changes:

* Show spinner in place of Retry after it's clicked
* Start polling for data after reshare
* Remove the spinner at the top of the modal to avoid shifting of the UI

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Enable the share status feature - `define( 'JETPACK_SOCIAL_HAS_SHARE_STATUS', true );`
* Goto the post editor for a post which has some failed shares or reshare posts to create a few entries
* Click on "View share history" in the publicize panel
* Click on Retry for a failed share
* Confirm that spinner is shown in place of Retry
* In the Network tab, confirm that polling is started
* Confirm that the table data refreshes when polling finishes.


https://github.com/user-attachments/assets/c6e58a96-6d5f-4f6c-9c58-a68f7f1a9a38
